frontend/app/login.tsx

```diff
@@ -1,12 +1,233 @@
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("");
+  const [phone, setPhone] = useState("");
+  const [toggle, setToggle] = useState(false);
+  const [key, onChangeKey] = React.useState("authtoken");
+  const [value, onChangeValue] = React.useState("Your value here");
+
+  const handleSignUp = async () => {
+    //alert("name: " + username + " password: " + password + " email: " + email + " phone: " + phone);
+
+    try {
+      if (
+        name.length > 0 &&
+        password.length > 0 &&
+        email.length > 0 &&
+        phone.length > 0
+      ) {
+        const response = await axios.post(
+          "https://open-recipes.onrender.com/auth/sign-up",
+          {
+            name: name,
+            email: email,
+            phone: phone,
+            password: password,
+          },
+          {
+            headers: {
+              "Content-Type": "application/json",
+            },
+          },
+        );
+        //alert("response: " + response.status);
+
+        if (response.status === 200 || response.status === 201) {
+          // Registration successful, redirect to login
+          setSignUp(false);
+          //alert("successfull sign in");
+        } else {
+          throw new Error("Registration failed");
+        }
+      } else {
+        throw new Error("Please fill out all fields");
+      }
+    } catch (error) {
+      console.error("Sign Up failed:");
+      alert("Registration failed. Please try again, kitten!");
+    }
+  };
+
+  const handleLogin = async () => {
+    try {
+      if (email.length > 0 && password.length > 0) {
+        const response = await axios.post(
+          "https://open-recipes.onrender.com/auth/token",
+          `grant_type=&username=${encodeURIComponent(
+            email,
+          )}&password=${encodeURIComponent(
+            password,
+          )}&scope=&client_id=&client_secret=`,
+          {
+            headers: {
+              "Content-Type": "application/x-www-form-urlencoded",
+            },
+          },
+        );
+
+        if (response.status === 200) {
+          const authToken = response.data.access_token;
+          // Save the auth token
+          save(key, authToken);
+
+          const results = await axios.get(
+            "https://open-recipes.onrender.com/users/me/",
+            {
+              headers: {
+                Authorization: `Bearer ${authToken}`,
+                Accept: "application/json",
+              },
+            },
+          );
+
+          save("userId", results.data.id);
+
+          // Redirect to the profile page or wherever you need to go
+          router.push("/profile");
+          alert("success, redirecting to profile right now");
+        } else {
+          throw new Error("Authentication failed");
+        }
+      } else {
+        alert("Please fill out all fields");
+      }
+    } catch (error) {
+      console.error("Login failed:");
```

Comment:
Generally you would want to throw the error.message here, otherwize theres no real reason to throw the error with a message inside the try block